packages/tag/test/mdx-provider.test.js

```diff
@@ -55,3 +55,22 @@ it('Should merge components when there is nested context', () => {
   // MDXTag picks up overridden component context
   expect(result).toMatch(/style="color:papayawhip"/)
 })
+
+it('Should merge components when there is nested context as a function', () => {
```

Comment:
Test name combined with what's under test makes it feel like the outer h1 component should still be used, but I don't think that's true?